lib/ts/recipe/emailpassword/api/implementation.ts

```diff
@@ -90,21 +243,167 @@ export default function getAPIImplementation(): APIInterface {
             | GeneralErrorResponse
         > {
             let email = formFields.filter((f) => f.id === "email")[0].value;
+            let userIdForPasswordReset: string | undefined = undefined;
+            let recipeUserId: string;
 
-            let user = await options.recipeImplementation.getUserByEmail({ email, userContext });
-            if (user === undefined) {
+            /**
+             * check if primaryUserId is linked with this email
+             */
+            let users = await listUsersByAccountInfo({
+                email,
+            });
+            if (users !== undefined) {
+                let primaryUser = users.find((u) => u.isPrimaryUser);
+                if (primaryUser !== undefined) {
```

Comment:
why have you flipped the conditions here in comparison to the diagram? In the diagram, we check for if email password used exists for the input email first, and then check if that user is a primary user or not - but here you are doing it the other way around.